File: treebolicOne/src/main/java/org/treebolic/one/TreebolicActivity.java
package org.treebolic.one;

import android.content.Context;
import android.content.Intent;
import android.widget.Toast;

import org.treebolic.TreebolicIface;

import androidx.annotation.NonNull;

/**
 * Treebolic standard activity
 *
 * @author Bernard Bou
 */
public class TreebolicActivity extends TreebolicSourceActivity
{
	// --Commented out by Inspection START (9/23/17 5:08 PM):
	//	/**
	//	 * Log tag
	//	 */
	//	private static final String TAG = "TreebolicA";
	// --Commented out by Inspection STOP (9/23/17 5:08 PM)

	// C O N S T R U C T O R

	public TreebolicActivity()
	{
		super(R.menu.treebolic);
	}

	// Q U E R Y

	@Override
	protected void query()
	{
		// sanity check
		if (this.providerName == null && this.source == null)
		{
			Toast.makeText(this, R.string.error_null_data, Toast.LENGTH_LONG).show();
			finish();
			return;
		}

		// query
		this.widget.init(this.providerName, this.source);
	}

	@Override
	protected void requery(@NonNull String source0)
	{
		this.source = source0.endsWith(".xml") ? source0 : source0 + ".xml";
		this.widget.reinit(this.source);
	}

	// I N T E N T

	/**
	 * Make Treebolic activity intent
	 *
	 * @param context      context
	 * @param providerName providerName class
	 * @param source       source
	 * @param base         base
	 * @param imageBase    image base
	 * @param settings     settings
	 * @param style        style
	 * @return intent
	 */
	@NonNull
	static public Intent makeTreebolicIntent(final Context context, final String providerName, final String source, final String base, final String imageBase, final String settings, final String style)
	{
		final Intent intent = new Intent(context, TreebolicActivity.class);
		intent.putExtra(TreebolicIface.ARG_PROVIDER, providerName);
		intent.putExtra(TreebolicIface.ARG_SOURCE, source);
		intent.putExtra(TreebolicIface.ARG_BASE, base);
		intent.putExtra(TreebolicIface.ARG_IMAGEBASE, imageBase);
		intent.putExtra(TreebolicIface.ARG_SETTINGS, settings);
		intent.putExtra(TreebolicIface.ARG_STYLE, style);
		return intent;
	}
}

File: treebolicOneSql/src/main/java/org/treebolic/one/sql/TreebolicActivity.kt
/*
 * Copyright (c) 2023. Bernard Bou
 */
package org.treebolic.one.sql

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.widget.Toast
import org.treebolic.TreebolicIface

/**
 * Treebolic standard activity
 *
 * @author Bernard Bou
 */
class TreebolicActivity : TreebolicSourceActivity(R.menu.treebolic) {

    // Q U E R Y

    override fun query() {

        // sanity check
        if (providerName == null && source == null) {
            Toast.makeText(this, R.string.error_null_data, Toast.LENGTH_LONG).show()
            finish()
            return
        }

        // query
        widget!!.init(providerName, source)
    }

    override fun requery(newSource: String?) {
        if (source != null) {
            val fields = source!!.split(",".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
            val newFields = newSource?.split(",".toRegex())?.dropLastWhile { it.isEmpty() }?.toTypedArray()
            if (newFields != null) {
                source = fields[0] + ',' + (if (newFields.size > 1) newFields[1] else newSource) + ','
            }
            Log.d(TAG, "New source: $newSource saved:$source")
        }
        restoring = true
        widget!!.reinit(newSource)
    }

    companion object {

        private const val TAG = "TreebolicA"

        // I N T E N T

        /**
         * Make Treebolic activity intent
         *
         * @param context      context
         * @param providerName providerName class
         * @param source       source
         * @param base         base
         * @param imageBase    image base
         * @param settings     settings
         * @param style        style
         * @param urlScheme    URL scheme
         * @param more         more data in bundle
         * @return intent
         */
        fun makeTreebolicIntent(context: Context?, providerName: String?, source: String?, base: String?, imageBase: String?, settings: String?, style: String?, urlScheme: String?, more: Bundle?): Intent {
            val intent = Intent(context, TreebolicActivity::class.java)
            intent.putExtra(TreebolicIface.ARG_PROVIDER, providerName)
            intent.putExtra(TreebolicIface.ARG_SOURCE, source)
            intent.putExtra(TreebolicIface.ARG_BASE, base)
            intent.putExtra(TreebolicIface.ARG_IMAGEBASE, imageBase)
            intent.putExtra(TreebolicIface.ARG_SETTINGS, settings)
            intent.putExtra(TreebolicIface.ARG_STYLE, style)
            intent.putExtra(TreebolicIface.ARG_URLSCHEME, urlScheme)
            intent.putExtra(TreebolicIface.ARG_MORE, more)
            return intent
        }
    }
}
